project/app/api/predict.py

```diff
@@ -149,35 +179,87 @@ async def predict_gas(item: GasItem):
     total = 0
     distance_in_region = split_by_region(item.coords)
 
+    resp = {}
+
     for i, distance in enumerate(distance_in_region['distances']):
         region = distance_in_region['regions'][i]
         miles = distance * meter_to_mile
-        regional_rate = region_gas_predictions(region, month, day, year)
-        total += (miles / mpg) * regional_rate
-
-        # print(region)
-        # print(miles)
-        # print(regional_rate)
 
-    return round(total, 2)
+        try:
+            # This try catch is here because I validated coordinates are in the
+            # contiguous USA by a simple bounding rectangle. It is possible to
+            # put locations within this bounding box, but outside of the USA
+            # proper. Like coordinates in the ocean, or just barely over the
+            # border. With more time, I'd hone my input validators to be more
+            # precise, but this works too.
+            # Example: Ottawa CA (-75.995000, 45.424721)
+            regional_rate = region_gas_predictions(region, month, day, year)
+        except:
+            detail = 'At least one coordinate lays outside the contiguous USA'
+            raise HTTPException(status_code = 422, detail = detail)
+        total += (miles / mpg) * regional_rate
 
+    resp['total'] = round(total, 2)
+    return resp
```

Comment:
Just some clarification here - originally predict_gas was returning a number and now it is returning a dictionary. We just have to make sure that whenever this function is called this should be taken into account.

Reply:
Good call. Uvicorn will convert the dict to a json object, so the user will actually get a json object back.

I did a little reading on the weekend, and I guess stylistically a single value json object is preferable to just a float or an int.  Docus page should help inform that usage.

Reply:
Oh gotcha, that makes sense. Thanks for clarifying.